UltraStar Play/Assets/Common/Graphics/BackgroundsDynamic/ThemeManager.cs

```diff
@@ -0,0 +1,428 @@
+                string replacement = $"{{ \"r\":{HexToFloatStr(match.Groups[1].Value, true)}, \"g\":{HexToFloatStr(match.Groups[2].Value, true)}, \"b\":{HexToFloatStr(match.Groups[3].Value, true)}, \"a\":{HexToFloatStr(match.Groups[4].Value, true)} }}";
+                input = input.Remove(match.Index + offset, match.Length).Insert(match.Index + offset, replacement);
+                offset += replacement.Length - match.Length;
+            });
+            offset = 0;
+            hexRgb.Matches(input).ForEach(match =>
+            {
+                string replacement = $"{{ \"r\":{HexToFloatStr(match.Groups[1].Value, true)}, \"g\":{HexToFloatStr(match.Groups[2].Value, true)}, \"b\":{HexToFloatStr(match.Groups[3].Value, true)}, \"a\":1.0 }}";
+                input = input.Remove(match.Index + offset, match.Length).Insert(match.Index + offset, replacement);
+                offset += replacement.Length - match.Length;
+            });
+
+            return input;
+        }
+
+        static string HexToFloatStr(string hex, bool singleDigit = false)
+        {
+            return (Convert.ToInt32(singleDigit ? $"{hex}{hex}" : hex, 16)/255.0f).ToString(CultureInfo.InvariantCulture);
+        }
+    }
+
+    [Serializable]
+    public class SongRatingIcons
+    {
+        public string toneDeaf;
+        public string amateur;
+        public string wannabe;
+        public string hopeful;
+        public string risingStar;
+        public string leadSinger;
+        public string superstar;
+        public string ultrastar;
+
+        Dictionary<string, Sprite> loadedSprites = new Dictionary<string, Sprite>();
+
+        public Sprite GetSpriteForRating(SongRating.ESongRating songRating)
+        {
+            switch (songRating)
+            {
+                case SongRating.ESongRating.ToneDeaf: return GetSprite(toneDeaf);
+                case SongRating.ESongRating.Amateur: return GetSprite(amateur);
+                case SongRating.ESongRating.Wannabe: return GetSprite(wannabe);
+                case SongRating.ESongRating.Hopeful: return GetSprite(hopeful);
+                case SongRating.ESongRating.RisingStar: return GetSprite(risingStar);
+                case SongRating.ESongRating.LeadSinger: return GetSprite(leadSinger);
+                case SongRating.ESongRating.Superstar: return GetSprite(superstar);
+                case SongRating.ESongRating.Ultrastar: return GetSprite(ultrastar);
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(songRating), songRating, null);
+            }
+        }
+
+        private Sprite GetSprite(string filePath)
+        {
+            if (string.IsNullOrEmpty(filePath))
+            {
+                return null;
+            }
+
+            if (loadedSprites.ContainsKey(filePath))
+            {
+                return loadedSprites[filePath];
+            }
+
+            string fullPath = $"{Application.dataPath}/../themes/{filePath}";
```

Comment:
- `Application.dataPath` points to the APK on Android. use `persistentDataPath` to load user defined files at runtime.
- We could copy default themes to persistentDataPath on startup, if the files do not exist yet.
   - Therefor, the default themes could be placed in a Resources folder